src/libp2p/Libp2p.Protocols.TLS/TLS.cs

```diff
@@ -0,0 +1,408 @@
+using System.Buffers;
+using System.Net;
+using System.Net.Security;
+using Nethermind.Libp2p.Protocols.Quic;
+
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Logging;
+using System.Security.Cryptography;
+using Nethermind.Libp2p.Core;
+using Multiformats.Address;
+using Multiformats.Address.Protocols;
+
+using System.Text;
+
+
+namespace Nethermind.Libp2p.Protocols;
+
+public class TlsProtocol : IProtocol
+{
+    public readonly ILogger<TlsProtocol>? _logger;
+    private readonly ECDsa _sessionKey;
+    public SslApplicationProtocol? LastNegotiatedApplicationProtocol { get; private set; }
+    private readonly List<SslApplicationProtocol> _protocols;
+
+    public TlsProtocol(ILoggerFactory? loggerFactory = null)
+    {
+        _logger = loggerFactory?.CreateLogger<TlsProtocol>();
+        _logger?.LogDebug("TlsProtocol instantiated");
+        _sessionKey = ECDsa.Create();
+    }
+
+    public string Id => "/tls/1.0.0";
+
+
+
+    public async Task ListenAsync(IChannel downChannel, IChannelFactory? channelFactory, IPeerContext context)
+    {
+        _logger?.LogDebug("Handling connection");
```

Comment:
Could you in general revise logs across the changes. The idea is to inform on trace level about every somehow significant step, report with debug level about very main steps and problems with handling remote peer incoming data